ParticleObjects/particleDungeon_class.f90

```diff
@@ -423,19 +428,175 @@ subroutine printToFile(self, name)
     integer(shortInt)                  :: i
 
     filename = trim(name)//'.txt'
-    open(unit = 10, file = filename, status = 'new')
+    open(unit = 10, file = filename)
 
     ! Print out each particle co-ordinate
     do i = 1, self % pop
-      write(10,'(8A)') numToChar(self % prisoners(i) % r), &
-                       numToChar(self % prisoners(i) % dir), &
-                       numToChar(self % prisoners(i) % E), &
-                       numToChar(self % prisoners(i) % G)
+      write(10,'(8A)') numToChar(self % prisoners(i) % r)!, &
+                       !numToChar(self % prisoners(i) % dir), &
+                       !numToChar(self % prisoners(i) % E), &
+                       !numToChar(self % prisoners(i) % G), &
+                       !numToChar(self % prisoners(i) % matIdx)
     end do
 
     ! Close the file
     close(10)
 
   end subroutine printToFile
 
+  !!
+  !! Prints given property of particles to screen
+  !!
+  !! Args:
+  !!   prop  [in] -> Particle property to be displayed
+  !!   nMax  [in] -> Maximum number of particles displayed
+  !!   total [in] -> Optional, if True then sum contributions of particles
+  !!                  and print for total
+  !!
+  !! Errors:
+  !!   fatalError if prop is invalid
+  !!
+  subroutine printToScreen(self, prop, nMax, total)
+    class(particleDungeon), intent(in)     :: self
+    character(*), intent(in)               :: prop
+    integer(shortInt), intent(in)          :: nMax
+    integer(shortInt)                      :: i,iMax
+    logical(defBool), intent(in), optional :: total
+    logical(defBool)                       :: totBool = .false.
```

Comment:
Initialising things in their declaration in a function is a little bit buggy we've found... Better to initialise it in the body of the function.